From b3525570929ba47c10d9d08696876c39487f7000 Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Thu, 16 Mar 2017 16:48:21 +0800
Subject: [PATCH] Fix test_format_xml with dot in path

When the path contains dot ".", replacing all dots will generate a non-exist result and raises a FileNotFoundError. Replacing only the last dot fixes this.
---
 test/mitmproxy/contentviews/test_xml_html.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/mitmproxy/contentviews/test_xml_html.py b/test/mitmproxy/contentviews/test_xml_html.py
index 899ecfd..2b0aee4 100644
--- a/test/mitmproxy/contentviews/test_xml_html.py
+++ b/test/mitmproxy/contentviews/test_xml_html.py
@@ -23,7 +23,7 @@ def test_format_xml(filename):
     path = data.path(filename)
     with open(path) as f:
         input = f.read()
-    with open(path.replace(".", "-formatted.")) as f:
+    with open("-formatted.".join(path.rsplit(".", 1))) as f:
         expected = f.read()
     tokens = xml_html.tokenize(input)
     assert xml_html.format_xml(tokens) == expected
